### PR TITLE
Fix item and wield meshes

### LINKED ISSUE #6536: Broken wield mesh and inventory rendering
On https://github.com/minetest/minetest/commit/75320e7e88ab5088a46c9e42c1e789cbdacb13b0 it appears the inventory_image (and wield_image?) are not drawing overlays.

screenshot_20171016_014023

The protection block normally displays this.  ~I think.~

Ps. I found this older screenshot, so it used to be on there: sss

### PR DESCRIPTION
**Pull request by numberZero**
Fixes #6536

### CODE REVIEW

**Comment by numberZero** [comment]
Inventory images work:
screenshot_20171107_002628

File: src/content_mapblock.cpp [hunk 1]
```diff
@@ -725,7 +725,7 @@ void MapblockMeshGenerator::drawGlasslikeFramedNode()
 			v3s16 n2p = blockpos_nodes + p + g_26dirs[i];
 			MapNode n2 = data->m_vmanip.getNodeNoEx(n2p);
 			content_t n2c = n2.getContent();
-			if (n2c == current || n2c == CONTENT_IGNORE)
+			if (n2c == current)
```
Comment (numberZero) [inline src/content_mapblock.cpp]: Otherwise `glasslike_framed` disappears from the inventory

**Comment by numberZero** [comment]
Done. @jastevenson303 can you test this?

**Comment by tobyplowy** [comment]
@numberZero Does this allow me to use models for items without using a block for the model?
Can I also animate it?

**Comment by numberZero** [comment]
@tobyplowy No, this only fixes the issue.
@juhdanad Can you test this with HW coloring, overlays, etc.?

**Comment by tobyplowy** [comment]
> @tobyplowy No, this only fixes the issue.

Ooohh that's sad can you please add it?

**Comment by jastevenson303** [comment]
@numberZero Yes, a quick test shows this fix working quite well!  Thanks!
screenshot_20171106_211713
I'll be certainly testing more tomorrow, and onward.

**Comment by numberZero** [comment]
I can, but not in this PR, this is a bugfix only.

Actually I plan to refactor the code a bit (oh yeah, it certainly needs that; still, not in this PR) and only then add any features.

File: src/wieldmesh.cpp [hunk 2]
```diff
@@ -531,30 +558,30 @@ void getItemMesh(Client *client, const ItemStack &item, ItemMesh *result)
 					break;
 				}
 				default: {
-					MeshMakeData mesh_make_data(client, false);
-					MapNode mesh_make_node(id, 255, 0);
-					mesh_make_data.fillSingleNode(&mesh_make_node);
-					MapBlockMesh mapblock_mesh(&mesh_make_data, v3s16(0, 0, 0));
-					mesh = cloneMesh(mapblock_mesh.getMesh());
-					translateMesh(mesh, v3f(-BS, -BS, -BS));
+					MeshMakeData mesh_make_data(client, false, false);
```
Comment (SmallJoker) [inline src/wieldmesh.cpp]: Could you please move this and the changed part above into a new function to
1) reduce the amount of indents
2) less code duplication?
I hope a few more checks are acceptable in terms of speed.

File: src/wieldmesh.cpp [hunk 3]
```diff
@@ -371,19 +374,43 @@ void WieldMeshSceneNode::setItem(const ItemStack &item, Client *client)
...
-					translateMesh(mesh, v3f(-BS, -BS, -BS));
-					postProcessNodeMesh(mesh, f, m_enable_shaders, true,
-						&m_material_type, &m_colors);
+					MeshMakeData mesh_make_data(client, false, false);
+					MeshCollector collector(false);
+					mesh_make_data.setSmoothLighting(false);
+					MapblockMeshGenerator gen(&mesh_make_data, &collector);
+					gen.renderSingle(id);
+					mesh = new scene::SMesh();
+					for (u8 layer = 0; layer < MAX_TILE_LAYERS; layer++)
```
Comment (nerzhul) [inline src/wieldmesh.cpp]: if i'm correct you can use a for rangebased loop there

File: src/wieldmesh.cpp (hunk 2 above)
Comment (numberZero) [reply src/wieldmesh.cpp]: Am I right that textures after 0th are not ever used (even if present) if shaders are disabled?

File: src/wieldmesh.cpp (hunk 3 above)
Comment (numberZero) [reply src/wieldmesh.cpp]: No, until I changed C array to std::array =) Hope that won’t break anything =)

File: src/wieldmesh.cpp (hunk 2 above)
Comment (SmallJoker) [reply src/wieldmesh.cpp]: According to https://github.com/minetest/minetest/blob/20a85d7/doc/lua_api.txt#L968 there can be multiple textures (tiles) for a mesh.

Comment (numberZero) [reply src/wieldmesh.cpp]: I mean textures after 0th in a single material.

**Comment by jastevenson303** [comment]
Testing; works for me: 
screenshot_20171110_200352

**Comment by numberZero** [comment]
So...?

**Comment by Fixer-007** [comment]
Will check it out and report.

**Comment by paramat** [comment]
This PR causes a bug https://github.com/minetest/minetest_game/issues/1959

**Pull request merged**